js/src/Ice/StringUtil.js

```diff
@@ -209,12 +209,26 @@ export class StringUtil {
         return hash;
     }
 
-    static toInt(s) {
-        const n = parseInt(s, 10);
-        if (isNaN(n)) {
-            throw new RangeError("conversion of `" + s + "' to int failed");
+    // Converts a string to an int32 number. We avoid using parseInt or the Number constructor because they do not
+    // detect overflow. This method throws a RangeError if the input is not a valid integer string or if the value
+    // is outside the 32-bit signed integer range.
+    static toInt32(s) {
+        // Reject empty strings and strings with leading or trailing whitespace.
+        if (s.length === 0 || s !== s.trim()) {
+            throw new RangeError(`conversion of '${s}' to int failed`);
+        }
+
+        try {
+            const b = BigInt(s);
```

Comment:
Does the BigInt constructor throw if not a number?

Reply:
yes

Reply:
Then do we really need to check above?

Reply:
Do you mean the check for the int32 range?

It depends of how we want to implement getPropertyAsInt, if we want it to return an int32 like in all other mappings we should. We can update the check to allow the Number safe integer range, ...